### DISCUSSION TITLE
Atomstack A5 lost its firmware (not connecting)

### DISCUSSION THREAD

**Discussion by MekoZeko** [arkypita/LaserGRBL#1273] [Q&A] [answered]
Hi. I used my laser engraver 2-3 days ago but today LaserGRBL didn't connecting to engraver. I tried to reinstall CH340 drivers and update LaserGRBL but they didn't work either. LaserGRBL session log : 

17.01.2021 01:03:36.461	Program     	------- LaserGRBL v4.0.5 [64bit] START -------
17.01.2021 01:03:39.737	SetStatus   	Machine status [Connecting]
17.01.2021 01:03:39.740	OpenCom     	Open COM4 @ 115200 baud  (UsbSerial)
17.01.2021 01:03:39.743	SendImmediate	Send Immediate Command [0x18]
17.01.2021 01:03:49.739	OpenCom     	Connection timeout!
17.01.2021 01:03:49.739	CloseCom    	Close COM4 [CORE]
17.01.2021 01:03:49.740	SetStatus   	Machine status [Disconnected]

CH340 Driver Settings : 
![image](https://user-images.githubusercontent.com/44033995/104824157-9e3f8600-5860-11eb-92d1-b7a27a6225a3.png)

Baud rate seems wrong but i didn't change anything before and it works like that. My machine is Atomstack A5.

Edit: I can see CH340 device on Device Manager. Its connecting to computer but not connecting to LaserGRBL

**Comment by arkypita** [comment]
Baud rate and settings inside COM configuration does not matter.
Just set 115200 in LaserGRBL

**Comment by arkypita** [comment]
![image](https://user-images.githubusercontent.com/8782035/104824262-734d3600-5850-11eb-9318-21217dd4bb22.png)

try setting hard and soft reset at connect

**Comment by arkypita** [comment]
Record an extended log (menu ? - activate extended log) and see if the engraver transmit anything

**Comment by MekoZeko** [comment]
[comlog.txt](https://github.com/arkypita/LaserGRBL/files/5825208/comlog.txt)

**Comment by arkypita** [comment]
Usually comlog look like this:

00005071	00000	com	Open COM4 @ 115200 baud 
00005133	00001	tx	[0x18]
00005141	00002	tx	[0x3F]
00005168	00003	rx	\r
00005176	00004	rx	Grbl 1.1f ['$' for help]\r
00005207	00005	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S|WCO:0.000,0.000,0.000>\r
00005348	00006	tx	[0x3F]
00005368	00007	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S|Ov:100,100,100>\r
00005548	00008	tx	[0x3F]
00005561	00009	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00005748	00010	tx	[0x3F]
00005761	00011	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00005948	00012	tx	[0x3F]
00005961	00013	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00006148	00014	tx	[0x3F]
00006159	00015	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r

If you see only tx and no rx it mean that the engraver board is not responding (died or bad firmware)

**Comment by arkypita** [comment]
I see your CH340 on COM4 but the log you upload talk about COM3

**Comment by MekoZeko** [comment]
I just change the port just to try its COM3 on device manager too

**Comment by MekoZeko** [comment]
Should I flash firmware?

**Comment by arkypita** [comment, marked as answer]
> Should I flash firmware?

If the board is not dead it could save you.

**Comment by arkypita** [comment]
Did you have any other computer to test with?

**Comment by MekoZeko** [comment]
I flashed the firmware to v1.1h original in LaserGRBL and it worked. Thanks for your help.

**Comment by arkypita** [comment]
you are welcome. I suggest to report your issue to atomstack, just for case history
